activerecord/lib/active_record/fixtures.rb

```diff
@@ -715,10 +716,13 @@ def ignored_fixtures=(base)
       # Loads the fixtures from the YAML file at +path+.
       # If the file sets the +model_class+ and current instance value is not set,
       # it uses the file value.
+
       def read_fixture_files(path)
-        yaml_files = Dir["#{path}/{**,*}/*.yml"].select { |f|
+        yaml_files = Dir["#{path}{.yml,/{**,*}/*.yml}"].select { |f|
           ::File.file?(f)
-        } + [yaml_file_path(path)]
+        }
+
+        raise ArgumentError, "No fixture files found for #{@name}" if yaml_files.empty?
```

Comment:
In the past implementation, `yaml_file_path` would always return a file name, so calling `FixtureSet::File.open(file)` would blow up if that file did not exist. In this new implementation we are globbing, so we will just get an empty array if the file doesn't exist. So I raise an equivalent error if no files are found.